internal/components/client.go

```diff
@@ -0,0 +1,56 @@
+package components
+
+import (
+	"github.com/amp-labs/connectors/common"
+	"github.com/amp-labs/connectors/providers"
+)
+
+type ClientComponent struct {
+	JSON *common.JSONHTTPClient
+	XML  *common.XMLHTTPClient
+	ProviderComponent
+}
+
+func NewClientComponent(
+	provider providers.Provider,
+	params common.Parameters,
+) (*ClientComponent, error) {
+	providerComponent, err := newProviderComponent(provider, params.Module, params.Workspace, params.Metadata)
+	if err != nil {
+		return nil, err
+	}
+
+	clientComponent := &ClientComponent{
+		JSON: &common.JSONHTTPClient{
+			HTTPClient: &common.HTTPClient{
+				Client: params.AuthenticatedClient,
+
```

Comment:
```suggestion
Base: providerComponent.ProviderInfo().BaseURL
```
This would remove field modifier after this struct initialization:
`clientComponent.JSON.HTTPClient.Base = providerComponent.ProviderInfo().BaseURL`